errai-codegen/src/main/java/org/jboss/errai/codegen/meta/MetaClassFactory.java

```diff
@@ -517,40 +528,16 @@ private static void addLookups(final Class cls, final MetaClass metaClass) {
 
   public static Class<?> loadClass(final String fullyQualifiedName) {
     try {
-      Class<?> cls = PRIMITIVE_LOOKUP.get(fullyQualifiedName);
-      if (cls == null) {
-        cls = Class.forName(fullyQualifiedName, false, Thread.currentThread().getContextClassLoader());
+      if (PRIMITIVE_LOOKUP.containsKey(fullyQualifiedName)) {
+        return PRIMITIVE_LOOKUP.get(fullyQualifiedName);
       }
-      return cls;
-    }
-    catch (final ClassNotFoundException e) {
-      final URL url = MetaClassFactory.class.getClassLoader()
-          .getResource(fullyQualifiedName.replace('.', '/') + ".java");
-
-      if (url != null) {
-        final File sourceFile = new File(url.getFile()).getAbsoluteFile();
-        if (sourceFile.exists()) {
-
-          final File directory = sourceFile.getParentFile();
-          final String packageName = fullyQualifiedName.substring(0, fullyQualifiedName.lastIndexOf('.'));
-          final String className = fullyQualifiedName.substring(fullyQualifiedName.lastIndexOf('.') + 1);
-
-          final String location = ClassChangeUtil.compileClass(directory.getAbsolutePath(),
-              packageName,
-              className,
-              RebindUtils.getTempDirectory() + "/hotload/");
-
-          try {
-            return ClassChangeUtil.loadClassDefinition(location,
-                packageName, className);
-          }
-          catch (final Exception e2) {
-            throw new RuntimeException("Could not load class: " + fullyQualifiedName, e2);
-          }
-        }
```

Comment:
@mbarkley This code was causing me trouble during development, so I deleted it. Since no tests were failing, I assumed it was ok to include this deletion on the PR. If you see any troubles that deleting this code might bring, please tell me and I'll put it back and fix whatever errors appear :)

Reply:
Great :)